catboost/jvm-packages/catboost4j-prediction/src/native/ai_catboost_CatBoostJNIImpl.cpp

```diff
@@ -92,8 +93,13 @@ static void HashCatFeatures(
         // NOTE: instead of C-style cast `dynamic_cast` should be used, but compiler complains that
         // `_jobject` is not a polymorphic type
         const auto jcatFeature = (jstring)jenv->GetObjectArrayElement(jcatFeatures, i);
+        CB_ENSURE(jcatFeature, "got null array element");
+        Y_SCOPE_EXIT(jenv, jcatFeature) {
+            jenv->DeleteLocalRef(jcatFeature);
```

Comment:
Are you sure this is really necessary? Isn't JNI suppose to discard all local references on exit from native method?

Reply:
It's not strictly necessary, but I think there's a maximum on the number of local references allowed per JNI call. I figured any kind of loop with user-supplied data can overflow that limit (I think it's 512 by default?) That said my only evidence of this is stories like https://github.com/assimp/assimp/issues/712